c7n/resources/account.py

```diff
@@ -1386,3 +1386,68 @@ def process(self, resources):
             client.put_public_access_block(
                 AccountId=r['account_id'],
                 PublicAccessBlockConfiguration=config)
+
+
+@filters.register('glue-security-config')
+class GlueEncryptionEnabled(MultiAttrFilter):
+    """Filter aws account by its glue encryption status and KMS key """
+
+    """:example:
+
+    .. yaml:
+
+      policies:
+        - name: glue-security-config
+          resource: aws.account
+          filters:
+            - type: glue-security-config
+                key: SseAwsKmsKeyId
+                value: alias/aws/glue
+
+    """
+    retry = staticmethod(QueryResourceManager.retry)
+
+    schema = {
+        'type': 'object',
+        'additionalProperties': False,
+        'properties': {
+            'type': {'enum': ['glue-security-config']},
+            'CatalogEncryptionMode': {'type': 'string'},
+            'SseAwsKmsKeyId': {'type': 'string'},
+            'ReturnConnectionPasswordEncrypted': {'type': 'boolean'},
+            'AwsKmsKeyId': {'type': 'string'}}}
+
+    annotation = "c7n:glue-security-config"
+    permissions = ('glue:GetDataCatalogEncryptionSettings',)
+
+    def validate(self):
+        attrs = set()
+        for key in self.data:
+            if key in ['CatalogEncryptionMode',
+                       'ReturnConnectionPasswordEncrypted',
+                       'SseAwsKmsKeyId',
+                       'AwsKmsKeyId']:
+                attrs.add(key)
+        self.multi_attrs = attrs
+        return super(GlueEncryptionEnabled, self).validate()
+
+    def get_target(self, resource):
+        if self.annotation in resource:
+            return resource[self.annotation]
+        client = local_session(self.manager.session_factory).client('glue')
+        encryption_setting = client.get_data_catalog_encryption_settings().get(
+            'DataCatalogEncryptionSettings')
+        resource[self.annotation] = encryption_setting.get('EncryptionAtRest')
```

Comment:
just set the whole annotation to encryption_setting instead of doing it piecemeal